File: LISTA 1/Exercício 5/Exercicio 5 (RPC)/Servidor05.py
from xmlrpc.server import SimpleXMLRPCServer
class RPC:
    _metodos_rpc = ['classificacao']
    def __init__(self, direcao):
        self._servidor = SimpleXMLRPCServer(direcao, allow_none=True)
        for metodo in self._metodos_rpc:
            self._servidor.register_function(getattr(self, metodo))

    def classificacao(self, idade):
        idade = float(idade)
        if idade<5:   
            return "Idade insuficiente para a competição"
        elif (idade >= 5 and idade <= 7):
            return "Categoria: infantil A"

        elif(idade >= 8 and idade <= 10):
            return "Categoria: infantil B"

        elif(idade >= 11 and idade <= 13):
            return "Categoria: juvenil A"
        elif(idade >= 14 and idade <= 17):
            return "Categoria: juvenil B"
        else:
            return "Categoria: Maiores de 18 anos"
    def iniciar_servidor(self):
        self._servidor.serve_forever()
if __name__ == '__main__' :
    rpc = RPC(('', 20064))
    print("Servidor RPC iniciado ... ")
    rpc.iniciar_servidor()

File: LISTA 1/Exercício 5/Exercicio 5 (Sockets)/Servidor05.java
import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.io.OutputStream;
import java.io.InputStream;
import java.net.ServerSocket;
import java.net.Socket;

public class Servidor05 {
	public static void main(String[] args) throws IOException {
		ServerSocket ss = new ServerSocket(7777);
		System.out.println("Servidor esperando por conexoes...");
		Socket socket = ss.accept(); 
		System.out.println("Conexao de " + socket + "!");

		InputStream inputStream = socket.getInputStream();
		DataInputStream dataInputStream = new DataInputStream(inputStream);
		
		OutputStream outputStream = socket.getOutputStream();
		DataOutputStream dataOutputStream = new DataOutputStream(outputStream);

		double idade = dataInputStream.readDouble();
		
		dataOutputStream.writeUTF(classificacao(idade));
		dataOutputStream.flush();
		
		dataOutputStream.close();
		
		System.out.println("Fechando Sockets.");
		ss.close();
		socket.close();
	}
	
	public static String classificacao(double idade){
		String resposta;
		if(idade < 5)
			resposta = "Idade insuficiente para a competição";
		else if(idade >= 5 && idade <= 7)
			resposta = "Categoria: infantil A";
		else if(idade >= 8 && idade <= 10)
			resposta = "Categoria: infantil B";
		else if(idade >= 11 && idade <= 13)
			resposta = "Categoria: juvenil A";
		else if(idade >= 14 && idade <= 17)
			resposta = "Categoria: juvenil B";
		else 
			resposta = "Categoria: Maiores de 18 anos";
		return resposta;
	}
}